### PR TITLE
test (MM-50964): Onboarding flow for newly installed Self-managed servers

### PR DESCRIPTION
**Pull request by stevemudie**
#### Summary

Initial onboarding for admins installing and getting started with a new Self-managed server.

A separate follow-up test case is on the way which will address the actual onboarding steps that appear in a list at the bottom of the LHS once the admin first lands on the server.

Related ticket: [MM-50964](https://mattermost.atlassian.net/browse/MM-50964)

### CODE REVIEW

File: data/test-cases/suite/onboarding/self-managed-onboarding.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Objective**
+
+- Verify the functionality of the Self-managed onboarding flow for the person installing the new server.
+
+Note: This test only addresses the different screens encountered during the onboarding and not the actual onboarding steps once the server is up and running. those steps are covered in a different test case.
```
Comment (lindalumitchell) [inline data/test-cases/suite/onboarding/self-managed-onboarding.md:39]:
```suggestion
Note: This test only addresses the different screens encountered during the onboarding and not the actual onboarding steps once the server is up and running. Those steps are covered in a different test case.
```

File: data/test-cases/suite/onboarding/self-managed-onboarding.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+1. Upon successful installation of a Self-managed server the first admin is presented with a page asking them to name their organization. Verify filling in a name allows the admin to continue onto the next step.
+
+![](https://raw.githubusercontent.com/mattermost/mattermost-test-management/main/data/asset/self-managed-onboarding/onboarding_org_name_screen.png)
+
+2. The next step of onboarding opens up the "What tools do you use?" screen. Verify that you can select as many or as few (including none) as you'd like and both the "Continue" or the "Skip" buttons allow you to move on.
+
+![](https://raw.githubusercontent.com/mattermost/mattermost-test-management/main/data/asset/self-managed-onboarding/onboarding_tools_screen.png)
+
+3. The next step opens up the "Invite your team members" page. Verify that clicking the "Copy Invite" button copies the URL to the clipboard and that clicking "Finish setup" moves you along to the "Launching you workspace now" transition page with an animation playing to idicate that it's not stuck in the process.
```
Comment (lindalumitchell) [inline data/test-cases/suite/onboarding/self-managed-onboarding.md:55]:
```suggestion
3. The next step opens up the "Invite your team members" page. Verify that clicking the "Copy Invite" button copies the URL to the clipboard and that clicking "Finish setup" moves you along to the "Launching your workspace now" transition page with an animation playing to indicate that it's not stuck in the process.
```

**Review by lindalumitchell — APPROVED** [review]
Just a couple very minor suggestions. LGTM!

File: data/test-cases/suite/onboarding/self-managed-onboarding.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+Note: This test only addresses the different screens encountered during the onboarding and not the actual onboarding steps once the server is up and running. Those steps are covered in a different test case.
+
+**Precondition**
+
+
+
+**Step 1**
+
+1. Upon successful installation of a Self-managed server the first admin is presented with a page asking them to name their organization. Verify filling in a name allows the admin to continue onto the next step.
```
Comment (marianunez) [inline data/test-cases/suite/onboarding/self-managed-onboarding.md:47]: Should we add the precondition that they have to complete the first admin account creation? This is the actual first step before the organization name

File: data/test-cases/suite/onboarding/self-managed-onboarding.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+![](https://raw.githubusercontent.com/mattermost/mattermost-test-management/main/data/asset/self-managed-onboarding/onboarding_transition_screen.png)
+
+4. Verify that the transition page moves automatically to the server workspace with the admin onboarding steps already expanded.
+
+![](https://raw.githubusercontent.com/mattermost/mattermost-test-management/main/data/asset/self-managed-onboarding/onboarding_product_landing_screen.png)
+
+**Expected**
+
+The end result is that the server is in place with any plugins that were selected in step 2 and it's ready to use.
```
Comment (marianunez) [inline data/test-cases/suite/onboarding/self-managed-onboarding.md:67]: We should also verify that a team was created with the name provided in the Company Name screen (step 1)

**Comment by saturninoabril** [comment]
Test key: MM-T5414 - https://github.com/mattermost/mattermost-test-management/blob/main/data/test-cases/suite/onboarding/MM-T5414.md

**Comment by stevemudie** [comment]
Thanks @saturninoabril!

While adding links to the traceability steps for [MM-T5414](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T5414) on Zephyr Scale I noticed that one of the screenshots is missing.

Specifically it's [this one](https://github.com/mattermost/mattermost-test-management/blob/main/data/asset/self-managed-onboarding/onboarding_transition_screen.jpeg).

Do you think it has anything to do with there being two images stacked one on top of the other?

https://github.com/mattermost/mattermost-test-management/blob/main/data/test-cases/suite/onboarding/MM-T5414.md?plain=1#L65

**Comment by saturninoabril** [comment]
I see and good catch Steve. Looking at the md file, it should be in `jpeg` but was reference as `png`. It's now corrected - https://github.com/mattermost/mattermost-test-management/commit/32a2bc50432356671ffc1a24482c635598275e47

**Pull request merged**